File: go/word-count/word_count.go
package wordcount

import (
	"regexp"
	"strings"
)

// Frequency records the number of times a word appears in a string.
type Frequency map[string]int

// WordCount counts the number of times a word appears in a string and returns a Frequency object containing the counts.
func WordCount(in string) Frequency {
	results := Frequency{}

	for _, word := range regexp.MustCompile(`[,\s]+`).Split(in, -1) {
		word = regexp.MustCompile(`[!&@$%^:.]+`).ReplaceAllString(word, "")
		word = regexp.MustCompile(`'$`).ReplaceAllString(word, "")
		word = regexp.MustCompile(`^'`).ReplaceAllString(word, "")
		word = strings.ToLower(word)
		if !regexp.MustCompile(`['a-z\d]+`).MatchString(word) { // Filter out blanks.
			continue
		}

		results[word]++
	}

	return results
}

File: python/word-count/word_count.py
import re

def word_count(phrase):
    words = {}
    for word in re.split(r'[ \t_,\n]+', phrase.lower()):
        word = word.rstrip(".:!&@$%^'")
        word = word.lstrip("'")
        if len(word) == 0:
            continue

        if word in words:
            words[word] += 1
        else:
            words[word] = 1
    return words
